### PR TITLE
gh-actions: add codespell workflow config

### PR DESCRIPTION
**Pull request by RandyMcMillan**

### CODE REVIEW

**Comment by RandyMcMillan** [comment]
Similar to the travis-ci config

###  NOTE:

---
This PR is built on top of #860 - this is the only way it'll pass the gh-actions config. (It is actually checking spelling)

---

Turn on Actions here --> https://github.com/lnbook/lnbook/actions

---

REF: https://github.com/RandyMcMillan/lnbook/actions

**Comment by RandyMcMillan** [comment]
This PR is built on top of #860 - this is the only way it'll pass the gh-actions config. (It is checking spelling)

File: preface.asciidoc [hunk 1]
```diff
@@ -161,8 +161,8 @@ Thank you all for supporting me throughout this journey.
 
 === Acknowledgments by Rene
 
-I want to thank the German education system through which I aquired the knowledge upon which my work builds. It is one of the greatest gifts I was given. 
-Similarly I want to thank the German public healthcare system and every person devoting their time into working within that industry. Your effort and endurance makes you my personal heros and I will never forget the help, patience and support I received when I was in need. 
+I want to thank the German education system through which I acquired the knowledge upon which my work builds. It is one of the greatest gifts I was given. 
+Similarly I want to thank the German public healthcare system and every person devoting their time into working within that industry. Your effort and endurance makes you my personal heroes and I will never forget the help, patience and support I received when I was in need. 
 Thanks goes to all the students I was allowed to teach and who engaged into interesting discussions and questions. From you I learnt the most.
```
Comment (RandyMcMillan) [inline preface.asciidoc:166]: While configuring this PR - additional spelling errors were detected.

File: contrib/channel-construction.asciidoc [hunk 2]
```diff
@@ -247,7 +247,7 @@ That is why we only explain this in the routing chapter and ask you to stay pati
 [NOTE]
 ====
 *TODO: Move this note to routing chapter?*
-HTLCS fullfil the following purposes:
+HTLCS fulfill the following purposes:
 1. Make a conditional payment.
```
Comment (RandyMcMillan) [inline contrib/channel-construction.asciidoc:251]: While configuring this PR - additional spelling errors were detected.

File: 10_onion_routing.asciidoc [hunk 3]
```diff
@@ -636,7 +636,7 @@ If Bob extracts and removes his hop payload, the remaining onion will not be 130
 
 Bob generates filler in a slightly different way than Alice, but following the same general principle.
 
-First, Bob *extends* the onion payload by 1300 bytes and filles them with +0+ values. Now the onion packet is 2600 bytes long, with the first half containing the data Alice sent and the next half containing zeroes. This operation is shown in <<bob_extends>>:
+First, Bob *extends* the onion payload by 1300 bytes and fills them with +0+ values. Now the onion packet is 2600 bytes long, with the first half containing the data Alice sent and the next half containing zeroes. This operation is shown in <<bob_extends>>:
```
Comment (RandyMcMillan) [inline 10_onion_routing.asciidoc:640]: While configuring this PR - additional spelling errors were detected.

File: contrib/path-finding.asciidoc [hunk 4]
```diff
@@ -308,7 +308,7 @@ This is the reason why arbitrary probing should be conducted with a fake, i.e. p
 With fake payment hashes the sender can probe concurrently as long as the sender has enough funds to pay for all the HTLCs.
 Successful probing does not guarantee a following successful payment.
 Assume a fake onion returns indicating that the payment hash was unknown to the recipient but otherwise the path has been possible.
-The sender now uses the same path to send the payment with the corrent payment hash.
+The sender now uses the same path to send the payment with the current payment hash.
 In the interim, the balance of a channel along the path changes rendering the path unworkable.
```
Comment (RandyMcMillan) [inline contrib/path-finding.asciidoc:312]: While configuring this PR - additional spelling errors were detected.

File: 04_node_client.asciidoc [hunk 5]
```diff
@@ -390,7 +390,9 @@ $ docker exec c-lightning cli getinfo
 
 We now have our first Lightning node running on a virtual network and communicating with a test Bitcoin blockchain. Later in this chapter we will start more nodes and connect them to each other to make some Lightning payments.
 
-In the next section we will also look at how to download, configure and compile c-lightning directly from the source code. This is an optional and advanced step that will teach you how to use the build tools and allow you to make modifications to c-lighting source code. With this knowledge you can write some code, fix some bugs, or create a plugin for c-lightning.
+In the next section we will also look at how to download, configure and compile c-lightning directly from the source
+code. This is an optional and advanced step that will teach you how to use the build tools and allow you to make
+modifications to c-lightning source code. With this knowledge you can write some code, fix some bugs, or create a plugin for c-lightning.
```
Comment (RandyMcMillan) [inline 04_node_client.asciidoc:396]: c-lighting -> c-lightning

File: 04_node_client.asciidoc [hunk 6]
```diff
@@ -390,7 +390,9 @@ $ docker exec c-lightning cli getinfo
 
 We now have our first Lightning node running on a virtual network and communicating with a test Bitcoin blockchain. Later in this chapter we will start more nodes and connect them to each other to make some Lightning payments.
 
-In the next section we will also look at how to download, configure and compile c-lightning directly from the source code. This is an optional and advanced step that will teach you how to use the build tools and allow you to make modifications to c-lighting source code. With this knowledge you can write some code, fix some bugs, or create a plugin for c-lightning.
```
Comment (RandyMcMillan) [inline 04_node_client.asciidoc:393]: c-lighting - c-lightning

**Comment by aantonop** [comment]
Current status is "IN PRODUCTION" and no PR's are accepted at this time. 

Sorry @RandyMcMillan I will have to close this until we are ready (in 2 months) to accept PRs for the second edition or print

**Pull request closed without merging**